streams/src/main/java/org/apache/kafka/streams/kstream/internals/foreignkeyjoin/SubscriptionWrapperSerde.java

```diff
@@ -60,6 +63,42 @@ public void setIfUnset(final SerdeGetter getter) {
             }
         }
 
+        @Override
+        public void configure(final Map<String, ?> configs, final boolean isKey) {
+            this.upgradeFromV0 = upgradeFromV0(configs);
+        }
+
+        private static boolean upgradeFromV0(final Map<String, ?> configs) {
+            final Object upgradeFrom = configs.get(StreamsConfig.UPGRADE_FROM_CONFIG);
+            if (!(upgradeFrom instanceof String)) {
```

Comment:
As above.

Reply:
Replaced with a not-null check in https://github.com/apache/kafka/commit/8ebbaaf38ee79642ccb322694de7556f260f93ca.